app/components/metadata-panel.tsx

```diff
@@ -49,32 +49,37 @@ type MetadataPanelState = {
   reset: () => void;
 };
 
-export const useMetadataPanelStore = create<MetadataPanelState>((set, get) => ({
-  open: false,
-  setOpen: (d: boolean) => {
-    set({ open: d });
-  },
-  toggle: () => {
-    set({ open: !get().open });
-  },
-  activeSection: "general",
-  setActiveSection: (d: MetadataPanelSection) => {
-    set({ activeSection: d });
-  },
-  selectedDimension: undefined,
-  setSelectedDimension: (d: DimensionMetadataFragment) => {
-    set({ selectedDimension: d });
-  },
-  clearSelectedDimension: () => {
-    set({ selectedDimension: undefined });
-  },
-  openDimension: (d: DimensionMetadataFragment) => {
-    set({ open: true, activeSection: "data", selectedDimension: d });
-  },
-  reset: () => {
-    set({ activeSection: "general", selectedDimension: undefined });
-  },
-}));
+export const createMetadataPanelStore = () =>
+  createStore<MetadataPanelState>((set, get) => ({
+    open: false,
+    setOpen: (d: boolean) => {
+      set({ open: d });
+    },
+    toggle: () => {
+      set({ open: !get().open });
+    },
+    activeSection: "general",
+    setActiveSection: (d: MetadataPanelSection) => {
+      set({ activeSection: d });
+    },
+    selectedDimension: undefined,
+    setSelectedDimension: (d: DimensionMetadataFragment) => {
+      set({ selectedDimension: d });
+    },
+    clearSelectedDimension: () => {
+      set({ selectedDimension: undefined });
+    },
+    openDimension: (d: DimensionMetadataFragment) => {
+      set({ open: true, activeSection: "data", selectedDimension: d });
+    },
+    reset: () => {
+      set({ activeSection: "general", selectedDimension: undefined });
+    },
+  }));
+
+const defaultStore = createMetadataPanelStore();
+
+export const MetadataPanelStoreContext = React.createContext(defaultStore);
```

Comment:
We could have `useMetadataPanelStore` here right ? This would keep the same API as before for consumers.

```
const useMetadataPanelStore = (selector) => {
	const store = useContext(MetadataPanelStoreContext)
	return useStore(store, selector)
}
```